### ISSUE TITLE
Mono projects not building

### ISSUE THREAD

**Issue by Paulloz** [godotengine/godot#15621]
**Godot version:** `master` / a26e6c8

**OS/device including version:** *Linux Mint 18.2*

**Issue description:**
Mono projects do not build anymore. Trying to build result to the following error message in the editor:
```
Failed to generate GodotSharp solution. Error: 47
```
And the following ones in Godot's logs:
```
ERROR: _generate_cs_property: Condition ' getter->return_type != setter->arguments.back()->get().type ' is true. returned: ERR_BUG
   At: modules/mono/editor/bindings_generator.cpp:1032.
ERROR: _generate_cs_type: Failed to generate property 'stream_peer' for class 'PacketPeerStream'
   At: modules/mono/editor/bindings_generator.cpp:815.
```
Seems to me it was introduced at 9b8e8b2.  
I looked at [bindings_generator.cpp](https://github.com/godotengine/godot/blob/958a15c7a7ddf7e4247ea73ce6cc2b63cd0c644b/modules/mono/editor/bindings_generator.cpp#L1032) and it seems that, when processing the `stream_peer` property in `PacketPeer` class, `getter->return_type` is `StreamPeer` while `setter.arguments.back()->get().type` is `Ref`. However my understanding of the code stops here.

As far as I understand, there might be the same issue with the `connection` property in the `HTTPClient`.

**Comment by novemberist** [comment]
Same here, though https://github.com/godotengine/godot/issues/12466 suggests, that it might have something to do with the docs (which have been updated with one of the latest commits)

**Comment by akien-mga** [comment]
Ping @bojidar-bg.

**Comment by bojidar-bg** [comment]
Ah, my bad. Can you confirm if changing `_set_stream_peer` to `set_stream_peer` in the binding fixes it?

**Comment by Paulloz** [comment]
Indeed it does fix the issue with `PacketPeer`, the same kind of error on other classes shows up though. I probably can make a list.

**Comment by Paulloz** [comment]
Here is the list of properties I changed before I could build again (with the current getter/setter type):
* `stream_peer` in `PackerPeer` (getter: `StreamPeer`, setter: `Ref`)
* `target_fps` in `_Engine` (getter: `int`, setter: `float`)

Would a PR be easier for you?

**Comment by akien-mga** [comment]
A PR would be great :)

**Comment by akien-mga** [comment]
Fixed by #15622.
